nomad/csi_endpoint.go

```diff
@@ -586,6 +589,57 @@ func (v *CSIVolume) volAndPluginLookup(namespace, volID string) (*structs.CSIPlu
 	return plug, vol, nil
 }
 
+// serializedControllerRPC ensures we're only sending a single controller RPC to
+// a given plugin if the RPC can cause conflicting state changes.
```

Comment:
I think we're good, but sharing here some of my thought process: For this to work, all such RPCs need to be run on a single server to block in memory.  Looking at what I think are the relevant RPCs, it seems like they are all(?) `v.srv.forward()`ed (to the leader).  I suppose this is what you meant by "with an allowance for losing state (ex. leadership transitions)" - i.e. this serialization could be broken by switching leaders?  and we can't reasonably avoid that caveat.

Reply:
Right, that's exactly it.